azerothcore/azerothcore-wotlk
# fix(Core/DBCLoader): Add handler for possibly unused DBC format data type

## Changes Proposed:
-  Adds `FT_NA_BYTE` to DBC handling from DB.

## Issues Addressed:

- Closes #4312 

## Tests Performed:

- Builds, did not crash on startup when running with contents in the tables described in the issue.

## How to Test the Changes:

1. Create a line in the three tables (charstartoutfit_dbc, skilllineability_dbc, spellitemenchantmentcondition_dbc) and fill them with blizzlike values. (Can be gotten from [here](https://github.com/wowgaming/client-data/releases/download/dbc_sql_v1/AzerothcoreDBCToSQL.zip) as the original issue suggests, or can be manually filled from the DBC itself or [WoW.tools](https://wow.tools/dbc/?dbc=charstartoutfit&build=3.3.5.12340).)
2. Start the server and test that you're getting the gear on spawn you should, you have the skills you should, and I don't know how to test the last one past the server not crashing on load.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
